internal/batches/executor.go

```diff
@@ -358,6 +358,19 @@ func (x *executor) do(ctx context.Context, task *Task) (err error) {
 		log.Close()
 	}()
 
+	if x.features.selectSearchOperator {
+		x.updateTaskStatus(task, func(status *TaskStatus) {
+			status.CurrentlyExecuting = "Resolving file matches..."
+		})
+		task.Repository.FileMatches, err = x.FetchFileMatches(task.Repository)
```

Comment:
I assume `task.Repository` is a clone for each task, right? Otherwise we'd have a race condition